runtime/autoload/provider/node.vim

```diff
@@ -83,6 +83,17 @@ function! provider#node#Detect() abort
     let pnpm_opts.job_id = jobstart('pnpm --loglevel silent root -g', pnpm_opts)
   endif
 
+  let bun_opts = {}
+  if executable('bun')
+    let bun_opts = deepcopy(s:NodeHandler)
+    let bun_opts.entry_point = '/node_modules/neovim/bin/cli.js'
+    let bun_config_dir = '/.bun'
+    let bun_default_path = $HOME . bun_config_dir . '/install' . '/global' . bun_opts.entry_point
+    if filereadable(bun_default_path)
+      return [bun_default_path, '']
+    endif
+  endif
```

Comment:
the priority/order: shouldn't bun come after npm, yarn, and pnpm? What do you think is the appropriate order of package managers being used among npm, yarn, pnpm, and bun --- say when they are all installed and provide global `node_modules` at different paths?

Reply:
I guess we could just check the ones that returns faster first (just like that comment says). I could run some benchmarks (with hyperfine) to get the priority order. We could also just check by popularity. If the user probably has npm installed doesn't make much sense check for the others.

The yarn stuff can also early return, we might want to do something about that.